pkg/model/crd.go

```diff
@@ -718,6 +718,119 @@ func (r *CRD) SortedFieldNames() []string {
+
+	// If there is an explicit tag field path mentioned inside generator config,
+	// retrieve it.
+	if resConfig := r.cfg.GetResourceConfig(r.Names.Original); resConfig != nil {
+		if tagConfig := resConfig.TagConfig; tagConfig != nil {
+			if tagConfig.Path != nil {
+				tagFieldPath = *tagConfig.Path
+			}
+		}
+	}
+	// verify that the tagFieldPath exists inside CRD
+	for fName, field := range r.Fields {
+		if strings.EqualFold(field.Path, tagFieldPath) {
+			return fName, nil
+		}
+	}
+	// If the tagFieldPath did not exist in CRD, return an error
+	return "", fmt.Errorf("tag field path %s does not exist inside %s"+
+		" crd", tagFieldPath, r.Names.Original)
+}
+
+// GetTagField return the model.Field representing the Tag field for CRD. If no
+// such field is found an error is returned.
+func (r *CRD) GetTagField() (*Field, error) {
+	if r.GetIgnoreTagging() {
+		return nil, nil
+	}
+	tagFieldName, err := r.GetTagFieldName()
+	if err != nil {
+		return nil, err
+	}
+	tagField := r.Fields[tagFieldName]
+	return tagField, nil
+}
+
+// GetTagKeyMemberName returns the member name which represents tag key.
+// The default value is "Key". This is only applicable for tag fields with
+// shape list of struct.If the tag field shape is not list of struct, an empty
+// string is returned.
+func (r *CRD) GetTagKeyMemberName() (keyMemberName string) {
+	tagField, _ := r.GetTagField()
+	if tagField == nil {
+		return keyMemberName
+	}
+	// TagKey member field will only be present when the tag field shape is
+	// list of struct
+	if isListOfStruct(tagField) {
+		keyMemberName = "Key"
+		if resConfig := r.Config().GetResourceConfig(r.Names.Original); resConfig != nil {
+			if tagsConfig := resConfig.TagConfig; tagsConfig != nil {
+				if tagsConfig.KeyMemberName != nil && *tagsConfig.KeyMemberName != "" {
+					keyMemberName = *tagsConfig.KeyMemberName
+				}
+			}
+		}
+	}
+	return keyMemberName
+}
+
+// GetTagValueMemberName returns the member name which represents tag value.
+// The default value is "Value". This is only applicable for tag fields with
+// shape list of struct.If the tag field shape is not list of struct, an empty
+// string is returned.
+func (r *CRD) GetTagValueMemberName() (valueMemberName string) {
+	tagField, _ := r.GetTagField()
+	if tagField == nil {
+		return valueMemberName
+	}
+	// TagValue member field will only be present when the tag field shape is
+	// list of struct.
+	if isListOfStruct(tagField) {
+		valueMemberName = "Value"
+		if resConfig := r.Config().GetResourceConfig(r.Names.Original); resConfig != nil {
+			if tagsConfig := resConfig.TagConfig; tagsConfig != nil {
+				if tagsConfig.ValueMemberName != nil && *tagsConfig.ValueMemberName != "" {
+					valueMemberName = *tagsConfig.ValueMemberName
+				}
+			}
+		}
+	}
+	return valueMemberName
+}
+
+// isListOfStruct method returns true is the shape of field is list of struct,
+// false otherwise
+func isListOfStruct(f *Field) bool {
```

Comment:
Can we make this a `Field` method instead? Ex: `func (f *Field) IsListOfStruct() bool {}`

Reply:
This was a very specific method for handling the tags and i did not wish to add it to Field struct.
Field struct has generic method to retrieve the field shape, Hence i left the private `isListOfStruct` close to other tag specific method for coherency.